### PR TITLE
Issue #4937: Drop CDATA support in javadoc grammar

### PR DESCRIPTION
**Pull request by pbludov**
Issue #4937 

Regression report https://pbludov.github.io/issue-4937/ (no difference).
Nobody uses CDATA in Javadoc.

### CODE REVIEW

File: src/main/resources/com/puppycrawl/tools/checkstyle/grammar/javadoc/JavadocLexer.g4 [hunk 1]
```diff
@@ -1,5 +1,35 @@
 lexer grammar JavadocLexer;
 
+//Please add new tokens only in the end of list! Otherwise you break compatibility!
+tokens {
```
Comment (pbludov) [inline src/main/resources/com/puppycrawl/tools/checkstyle/grammar/javadoc/JavadocLexer.g4]: This need to keep the order.

File: src/test/java/com/puppycrawl/tools/checkstyle/grammar/javadoc/GeneratedJavadocTokenTypesTest.java [hunk 2]
```diff
@@ -46,7 +46,6 @@
     public void testTokenNumbers() {
         assertEquals(1, JavadocParser.LEADING_ASTERISK, MSG);
         assertEquals(2, JavadocParser.HTML_COMMENT_START, MSG);
-        assertEquals(3, JavadocParser.CDATA, MSG);
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/grammar/javadoc/GeneratedJavadocTokenTypesTest.java]: @romani Should this remain in?

**Comment by rnveach** [comment]
Pitest is missing from the CI.

**Comment by pbludov** [comment]
> and needs a project that is known to have CDATA.

I'm afraid there are no such projects. The Javadoc tool reports an error for CDATA.

**Comment by rnveach** [comment]
> I'm afraid there are no such projects.

I went on to saying can we add an example to checkstyle?

>  The Javadoc tool reports an error for CDATA.

Can you provide more details? Is this without your change or with it? What is the exact error?

**Comment by pbludov** [comment]
> > I'm afraid there are no such projects.
> 
> I went on to asking can we add an example to checkstyle?

will do.

> Can you provide more details? Are you referring to the shell script or checkstyle? Is this without your change or with it? What is the exact error?

The error described in the issue [description](https://github.com/checkstyle/checkstyle/issues/4937#issue-249729262):
```
Generating .\test2.html...
test2.java:2: error: malformed HTML
 * This is <![CDATA[sdas]]>
```
Once this PR will be merged, the Checkstyle parser will treat `<![CDATA[sdas]]>` as a regular text.
I'll try to make a test case with example.

**Comment by pbludov** [comment]
> I went on to asking can we add an example to checkstyle?

@rnveach done. Please review.

File: src/test/java/com/puppycrawl/tools/checkstyle/grammar/javadoc/GeneratedJavadocTokenTypesTest.java (hunk 2 above)
Comment (rnveach) [reply src/test/java/com/puppycrawl/tools/checkstyle/grammar/javadoc/GeneratedJavadocTokenTypesTest.java:50]: @pbludov If field is still in JavadocParser, then we need to keep this check in.

Comment (pbludov) [reply src/test/java/com/puppycrawl/tools/checkstyle/grammar/javadoc/GeneratedJavadocTokenTypesTest.java:50]: done.

**Pull request merged**